src/gui/manager.cpp

```diff
@@ -0,0 +1,192 @@
+/*
+Minetest
+Copyright (C) 2023 v-rob, Vincent Robinson <robinsonvincent89@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "gui/manager.h"
+
+#include "debug.h"
+#include "log.h"
+#include "settings.h"
+#include "client/client.h"
+#include "client/renderingengine.h"
+#include "client/tile.h"
+#include "util/serialize.h"
+
+namespace ui
+{
+	bool testShift(u32 &bits)
+	{
+		bool test = bits & 1;
+		bits >>= 1;
+		return test;
+	}
```

Comment:
Please explain the purpose of the `if (testShift(set_mask))` pattern you're employing. I suppose you're reading an u32 determing which fields are set at the beginning of the stream, and only reading the fields which are set?

Also, where in the code are the streams you're deserializing here serialized?